presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java

```diff
@@ -680,7 +727,77 @@ public ConnectorPageSource createPageSource(
                 table.getPredicate(),
                 splitContext.isCacheable());
 
-        return new IcebergPageSource(icebergColumns, partitionKeys, dataPageSource, session.getSqlFunctionProperties().getTimeZoneKey());
+        Optional<RowPredicate> deletePredicate = deleteFilters.stream()
+                .map(filter -> filter.createPredicate(regularColumns))
+                .reduce(RowPredicate::and);
+
+        return new IcebergPageSource(icebergColumns, partitionKeys, dataPageSource, deletePredicate, session.getSqlFunctionProperties().getTimeZoneKey());
+    }
+
+    private List<DeleteFilter> readDeletes(ConnectorSession session, SchemaTableName tableName, Schema schema, String dataFilePath, List<DeleteFile> deleteFiles)
+    {
+        Slice targetPath = utf8Slice(dataFilePath);
+        List<DeleteFilter> filters = new ArrayList<>();
+        LongBitmapDataProvider deletedRows = new Roaring64Bitmap();
+
+        IcebergColumnHandle deleteFilePath = IcebergColumnHandle.create(DELETE_FILE_PATH, typeManager);
+        IcebergColumnHandle deleteFilePos = IcebergColumnHandle.create(DELETE_FILE_POS, typeManager);
+        List<IcebergColumnHandle> deleteColumns = ImmutableList.of(deleteFilePath, deleteFilePos);
+        TupleDomain<IcebergColumnHandle> deleteDomain = TupleDomain.fromFixedValues(ImmutableMap.of(deleteFilePath, NullableValue.of(VARCHAR, targetPath)));
+
+        for (DeleteFile delete : deleteFiles) {
+            if (delete.content() == POSITION_DELETES) {
+                try (ConnectorPageSource pageSource = openDeletes(session, delete, tableName, deleteColumns, deleteDomain)) {
+                    readPositionDeletes(pageSource, targetPath, deletedRows);
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
```

Comment:
It seems like if this happens, it's kind of the same thing as it happening to the data file.  I wonder if we should do something like this instead 
```suggestion
                    throw new PrestoException(ICEBERG_CANNOT_OPEN_SPLIT, "Cannot open Iceberg delete file", e);
```